packages/bloc_test/lib/src/bloc_test.dart

```diff
@@ -78,30 +98,31 @@ import 'package:test/test.dart' as test;
 ///
 /// ```dart
 /// blocTest(
-///  'emits [StateA, StateB] when MyEvent is added',
-///  build: () => MyBloc(),
+///  'emits [StateB] when MyEvent is added',
+///  build: () async => MyBloc(),
 ///  act: (bloc) => bloc.add(MyEvent()),
-///  expect: [isA<StateA>(), isA<StateB>()],
+///  expect: [isA<StateB>()],
 /// );
 /// ```
 @isTest
 void blocTest<B extends Bloc<Event, State>, Event, State>(
   String description, {
-  @required B Function() build,
-  @required Iterable expect,
+  @required Future<B> Function() build,
   Future<void> Function(B bloc) act,
   Duration wait,
-  Future<void> Function() verify,
+  int skip = 1,
+  Iterable expect,
+  Future<void> Function(B bloc) verify,
 }) {
   test.test(description, () async {
-    final bloc = build();
+    final bloc = await build();
     final states = <State>[];
-    final subscription = bloc.listen(states.add);
+    final subscription = bloc.skip(skip).listen(states.add);
     await act?.call(bloc);
     if (wait != null) await Future.delayed(wait);
     await bloc.close();
     test.expect(states, expect);
```

Comment:
Missing null check for the no-longer `@required` `expect` parameter.

Reply:
good catch 👍